src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxNode.cs

```diff
@@ -285,15 +287,15 @@ public DirectiveTriviaSyntax GetFirstDirective(Func<DirectiveTriviaSyntax, bool>
         /// <summary>
         /// Gets the last directive of the tree rooted by this node.
         /// </summary>
-        public DirectiveTriviaSyntax GetLastDirective(Func<DirectiveTriviaSyntax, bool> predicate = null)
+        public DirectiveTriviaSyntax? GetLastDirective(Func<DirectiveTriviaSyntax, bool>? predicate = null)
         {
             foreach (var child in this.ChildNodesAndTokens().Reverse())
             {
                 if (child.ContainsDirectives)
                 {
                     if (child.IsNode)
                     {
-                        var d = child.AsNode().GetLastDirective(predicate);
```

Comment:
Do you think it's worth adding an extension method for this pattern with an out?

Reply:
Debated it, but Jared didn't introduce such pattern for most syntax changes he made. I'll stay with existing pattern.

---
In reply to: [405830860](https://github.com/dotnet/roslyn/pull/43025#discussion_r405830860) [](ancestors = 405830860)

Reply:
I definitely didn't use that pattern when I was first exploring annotating syntax. I was more getting used to the feature and didn't wan to change as much around it while I was annotating. Now though I'm more comfortable with that type of change and if we have a handy helper would opt for using it.

Reply:
We don't have a helper yet. Would rather keep that separate.

---
In reply to: [414137782](https://github.com/dotnet/roslyn/pull/43025#discussion_r414137782) [](ancestors = 414137782)

Reply:
Oh, actually, we do have `AsNode(out ...)`. I'll use that

---
In reply to: [414179069](https://github.com/dotnet/roslyn/pull/43025#discussion_r414179069) [](ancestors = 414179069,414137782)